grr/server/grr_response_server/gui/ui/components/client_details/entry_history_dialog/entry_history_dialog.ts

```diff
@@ -0,0 +1,54 @@
+import {Component, Inject} from '@angular/core';
+import {MAT_DIALOG_DATA} from '@angular/material/dialog';
+import {Client} from '@app/lib/models/client';
+
+export type EntryType =
+    'primitive'|'timestamp'|'size'|'user-list'|'interface-list'|'volume-list';
+
+export interface EntryHistoryDialogParams {
+  path: string;
+  type: EntryType;
+  clientVersions: Client[];
+}
+
+interface EntryHistoryTableRow {
+  time: Date;
+  version: any;
```

Comment:
Don't use `any` - use more precise types.

Reply:
I agree that having `any` as type is not preferred. Here `version` could be any kind of object (Date, string, number, StorageVolume, NetworkInterface, etc.). Having a custom type for this is more rigid and must be changed when a new custom type will define the property. Should we change anyway?

Reply:
Use generics:
```
interface EntryHistoryTableRow<T> {
  ....
  version: T;
```